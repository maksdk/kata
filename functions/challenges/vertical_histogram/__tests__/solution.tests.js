// @ts-check
/* eslint-disable no-console */

import fakerator from 'fakerator';
import displayHistogram from '../solution.js';

const getRandomFn = (seed) => {
  fakerator().seed(seed);
  return () => fakerator().random.number(1, 6);
};

test.each([32, 100, 125, 210, 9, 10, 13])(
  'test histogram with rounds count %s',
  (roundsCount) => {
    console.log = jest.fn();
    const rollDie = getRandomFn(roundsCount);
    displayHistogram(roundsCount, rollDie);
    const actual = console.log.mock.calls.join('\n');
    expect(actual).toMatchSnapshot();
  },
);