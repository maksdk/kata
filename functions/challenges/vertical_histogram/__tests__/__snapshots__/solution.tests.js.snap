// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test histogram with rounds count 9 1`] = `""`;

exports[`test histogram with rounds count 10 1`] = `""`;

exports[`test histogram with rounds count 13 1`] = `""`;

exports[`test histogram with rounds count 32 1`] = `""`;

exports[`test histogram with rounds count 100 1`] = `""`;

exports[`test histogram with rounds count 125 1`] = `""`;

exports[`test histogram with rounds count 210 1`] = `""`;
